python/services/dataservice/dmod/dataservice/initial_data_adder_impl.py

```diff
@@ -110,15 +112,15 @@ class FromPartialRealizationConfigAdder(InitialDataAdder):
     # TODO: centrally define this somewhere else
     _REAL_CONFIG_FILE_NAME = 'realization_config.json'
 
-    def __init__(self, job: Job, all_dataset_managers: Dict[DatasetType, DatasetManager], *args, **kwargs):
+    def __init__(self, job: Job, dataset_manager_collection: DatasetManagerCollection, *args, **kwargs):
         """
 
         Parameters
         ----------
         job : Job
             The job requiring a realization config dataset, which must be an ngen-related job.
-        all_dataset_managers : Dict[DatasetType, DatasetManager]
-            Dictionary of all dataset managers currently available to the data service, keyed by dataset type.
+        dataset_manager_collection: DatasetManagerCollection
+            Collection of DatasetManager objects and their associated DatasetType.
```

Comment:
Did you mean to generalize this param/attribute to no longer necessarily contain all managers?  That was my first impression, although I could see interpreting things either way. 

If "yes," should we add some more graceful/informative error handling if a necessary manager is left out (alternatively, should/do we make sure that all the right managers are included; this seems like it'd be harder to do unless we go back to the "all" case)?  Possibly a custom exception to represent this and better communicate it up the stack? 

If "no", should we try to do something more to ensure the passed param really contains all managers (e.g., singleton-ize the service manager and/or its _DatasetManagerCollection_)?

Assuming we feel like something more needs to be done along one of these lines, we don't have to do it now, as long as we track it with another issue.

Reply:
> Did you mean to generalize this param/attribute to no longer necessarily contain all managers? That was my first impression, although I could see interpreting things either way.

Great question. No, I just renamed it so it would be named consistent across the codebase. 



> If "no", should we try to do something more to ensure the passed param really contains all managers (e.g., singleton-ize the service manager and/or its _DatasetManagerCollection_)?

Maybe? I think the service is self contained enough at the moment that it doesn't merit adding that level of complexity. If this were intended to be a library that would be a different story. I'm persuadable though. Thoughts?

Reply:
That'a fair assessment.  I would like to have it be a little more clear if/when these need to be "all managers." but I don't want to hold this up.  I've opened #554 to take a closer look at things and add some documentation, but we can address that later.